### ISSUE TITLE
op-node stuck after L1 node switch

### ISSUE THREAD

**Issue by crebsy** [ethereum-optimism/optimism#6851]
**Describe the bug**
I'm running bedrock stack on mainnet with the dockerized `op-node:v1.1.1`
In the last two days, I lost network connection to the configured L1 node.
Today, I've switched the L1 URL and restarted op-node.
Now the op-node is stuck in a loop traversing back around 2 days of L1 blocks over and over again.
```
...
t=2023-08-17T03:45:25+0000 lvl=info msg="Walking back L1Block by hash"       curr=0x731ca1d47d67669803bf14e585db34fc55f7b218119ecc0dbc31851a7d7fe979:17927746 next=0x551ed54dbda8d7ff705e9f0c46c9f72cf58f217e3c2646af1cd27ec91de77657:17927745 l2block=0x6e2bf3aad26cbde7153017833634149da3fbe438b2eb79341c66fa33d526aea2:108297243
t=2023-08-17T03:45:25+0000 lvl=info msg="Walking back L1Block by hash"       curr=0x551ed54dbda8d7ff705e9f0c46c9f72cf58f217e3c2646af1cd27ec91de77657:17927745 next=0x7e3992df851edcb60d52608879eaa238f6420f396f305af0a0623e5f862dc7b1:17927744 l2block=0x6a72668599ba100f47cac1e7827ef3909e4935dd0666357ac7d498eccad4e8b4:108297237
t=2023-08-17T03:45:25+0000 lvl=info msg="Walking back L1Block by hash"       curr=0x7e3992df851edcb60d52608879eaa238f6420f396f305af0a0623e5f862dc7b1:17927744 next=0x3c9e4e966465efb8dee9086c7ab98d2e77dba6332163ef704a38e7cbe28fe0ef:17927743 l2block=0xbc957b7eb9e142405c5be4366208b0e1937d2e70e7d2202bd4a0f481123a586c:108297231
t=2023-08-17T03:45:25+0000 lvl=info msg="Walking back L1Block by hash"       curr=0x3c9e4e966465efb8dee9086c7ab98d2e77dba6332163ef704a38e7cbe28fe0ef:17927743 next=0xe148f6c7220bf5a9db1a7ae01a5726be40f2d5f378aa9bba1a761cd6eec8555e:17927742 l2block=0xf1f9cdb6e98f22771d0efed5118836cdedd8311f56630ec5627317b7ab04a56a:108297225
t=2023-08-17T03:45:25+0000 lvl=info msg="Walking back L1Block by hash"       curr=0xe148f6c7220bf5a9db1a7ae01a5726be40f2d5f378aa9bba1a761cd6eec8555e:17927742 next=0x4bff1e62b1727274627cbc82e28115b35a74ba1e2a84774c81b692ce0acae55c:17927741 l2block=0x40a5a34c6458e5a63a92458f7832af0477ebec2bdec5448a1ba297fde93f39b8:108297219
...
```

I've seen a few other logs indicating a reorg problem:
```
t=2023-08-17T03:01:06+0000 lvl=warn msg="Derivation process temporary error" attempts=1 err="stage 0 failed resetting: temp: failed to find the L2 Heads to start from: reorg is too deep: traversed back to L2 block 0x80e91068369632491fe9c34005b434ec268daf6fc5a0e72a4e96e610affa565d:108198044, but too deep compared to previous unsafe block 0x3a0ca33bfeff0d153f8b73178eb800617d5f02efb9ecd829f71df0ce6d830715:108306767"
t=2023-08-17T03:16:00+0000 lvl=warn msg="Derivation process temporary error" attempts=2 err="stage 0 failed resetting: temp: failed to find the L2 Heads to start from: reorg is too deep: traversed back to L2 block 0x80e91068369632491fe9c34005b434ec268daf6fc5a0e72a4e96e610affa565d:108198044, but too deep compared to previous unsafe block 0x3a0ca33bfeff0d153f8b73178eb800617d5f02efb9ecd829f71df0ce6d830715:108306767"
t=2023-08-17T03:30:00+0000 lvl=warn msg="Derivation process temporary error" attempts=3 err="stage 0 failed resetting: temp: failed to find the L2 Heads to start from: reorg is too deep: traversed back to L2 block 0x80e91068369632491fe9c34005b434ec268daf6fc5a0e72a4e96e610affa565d:108198044, but too deep compared to previous unsafe block 0x3a0ca33bfeff0d153f8b73178eb800617d5f02efb9ecd829f71df0ce6d830715:108306767"
```

**Expected behavior**
op-node should self-recover

Is there anything that can be done to recover the node ?

**Comment by 0xalex88** [comment]
In my case I've upgraded from 1.1.1 to 1.1.4 and instead of throwing that `reorg is too deep` and looping it showed `Hit finalized L2 head, returning immediately` (when it reached l1 block 0) and then it resumed working properly

**Comment by folowing** [comment]
> In my case I've upgraded from 1.1.1 to 1.1.4 and instead of throwing that `reorg is too deep` and looping it showed `Hit finalized L2 head, returning immediately` (when it reached l1 block 0) and then it resumed working properly

thanks, it works like a charm

**Comment by DaveWK** [comment]
Unfortunately my node continued to loop on v1.1.4 back down to zero over and over again. I had previously been using a Nethermind node for the L1, and in bedrock had set the "nethermind" node type. I was able to point it at a go-ethererum node and changed the "type" parameter to "basic" and now it is catching back up again.

**Comment by avi-piertwo** [comment]
> Unfortunately my node continued to loop on v1.1.4 back down to zero over and over again. I had previously been using a Nethermind node for the L1, and in bedrock had set the "nethermind" node type. I was able to point it at a go-ethererum node and changed the "type" parameter to "basic" and now it is catching back up again.

I have been using "nethermind" stack instead of "geth" and the sync just stops from time to time at certain blocks. Is it mandatory to run a Geth node for the l1 URL after the bedrock upgrade?

Thanks. 🙏

**Comment by DaveWK** [comment]
I am using a nethermind node, but I am using the "basic" type. Something seems broken with the nethermind enhancements

**Comment by avi-piertwo** [comment]
> I am using a nethermind node, but I am using the "basic" type. Something seems broken with the nethermind enhancements

I have been digging into this for some time, but we are running a _pruned_ Nethermind nodes stack which might have been causing the issue. Need to test this out with a new non-pruned stack to revert with better answers.

Did you happen to run a non-pruned Neth stack and still see the issue?
I am assuming you might have also tried "_nethermind_" as the [type](https://github.com/ethereum-optimism/optimism/blob/0025fff177d656fc14d64c174eef3b69a03413dc/op-node/sources/receipts.go#L281) to test your setup. If you use "_basic_" it runs through a list of eth calls to figure out which will suit the stack you are running which to me increases latency.

**Comment by DaveWK** [comment]
I have tried with a pruned and "archive" node with the "nethermind" type, wihch run into the same issue. As I understand it, nethemind has some non-standard rpc paths which I believe optimism is attempting to use to retrieve and decode relevant l2 transactions faster. I am able to use the rpc from nethermind with "basic" and I am able to keep up with the network on my l2, while not dealing with the awful instability and corruption issues associated with geth's usage of the goleveldb module. I'd be interested in seeing the metrics on how much faster the "nethermind" type is supposed to be, but it doesn't seem from my perspective like the integration tests are sufficient to prove reliability or performance improvements.,

**Comment by Rogalek** [comment]
I have the same problem - my node get stucked and its `Walking back L1Block by hash` for one week already.

My case is that space in my disk volume got 100%, so I had to extend volume. I had to turn off op-geth and op-node, and had to delete optimism directory because I could not ran any command in terminal without any space in terminal.

Now I installed everything again and ran nodes and this is happening for one week already.

I am worried that it get stacked because few days ago block ID was lesser than today:
```
Jan 05 09:02:19 my-server bash[4347]: t=2024-01-05T09:02:19+0000 lvl=info msg="Walking back L1Block by hash"        curr=0x8b00822ccf06c436c8881300e153ab6714fe161b0e158c6337bb5790a05a97b7:18264178 next=0x3bc0825482846f5bdf091de5abfbee1460897ba655e1c6a58fc71e066cea69ba:18264177 l2block=0x6ad96a63861c6415215ac63d552bb618d3a615c93f68186efe5415d266769c3b:110333528
Jan 05 09:02:19 my-server bash[4347]: t=2024-01-05T09:02:19+0000 lvl=info msg="Walking back L1Block by hash"        curr=0x3bc0825482846f5bdf091de5abfbee1460897ba655e1c6a58fc71e066cea69ba:18264177 next=0xeddf92494fceffff0fbef9b08fbbf90ef72062301080fcad585ba73649a7d79f:18264176 l2block=0x185672d17a8ace9eda3ecd1dda22b794ee0bd569f2032680b2b7143fd2de0eb4:110333522
```

And today:
```
Jan 11 11:02:48 my-server bash[4347]: t=2024-01-11T11:02:48+0000 lvl=info msg="Walking back L1Block by hash"        curr=0x98437d38e000a415b5d80169533ed0e1d046b7f537ca4249e54b862f803cad0e:18548343 next=0xd9d2b47334cfda62089bfa827e213e4266faec695d7129f2f40ce8757e3c4aaf:18548342 l2block=0xf3aee3538ae5a7cc802a84d8fdefa64b75088ffd1e1e5d694ba18d37b365aba0:112051414
Jan 11 11:02:48 my-server bash[4347]: t=2024-01-11T11:02:48+0000 lvl=info msg="Walking back L1Block by hash"        curr=0xd9d2b47334cfda62089bfa827e213e4266faec695d7129f2f40ce8757e3c4aaf:18548342 next=0xb41726190ffd3d40529feaab042d82a68beca68f83faf69ce637658e0721e14d:18548341 l2block=0xe9b2f96c8148b91965f5f5eea652cf1ca4c491aa00281d2186b9c54a5f797955:112051403
```

Please help!

**Comment by alexqrid** [comment]
Also faced almost the same issue with [`opbnb`](https://github.com/bnb-chain/opbnb/issues/112)

**Comment by PeaStew** [comment]
Same issue, all I did was stop and start the node, no changes

**Comment by trianglesphere** [comment]
Hi all, I am closing this issue because the first reported case was fixed in op-node 1.1.4. The minimum supported version for mainnet is 1.3.2. If you are running into this issue on a modern version of op-node, please open a new issue with information about the node versions & network you're running.
